### DISCUSSION TITLE
can we create diff master encrypted key for etch user in minio so KES will get the key based on the user ?

### DISCUSSION THREAD

**Discussion by aviscode** [minio/minio#13118] [Q&A] [answered]
can we create diff master encrypted key for each user in minio so KES will get the key based on the user? 
i will explain if we have 2 users with diff policies so one user can't see the data of the user i will like each user to have diff encryption that means the in vault i will have a key for each user, can minio see which user is doing the put/get and based of it to get the master key in the KES?

**Comment by klauspost** [comment, marked as answer]
You can set up different keys per bucket (see [`mc encrypt set`](https://docs.min.io/minio/baremetal/reference/minio-cli/minio-mc/mc-encrypt.html#mc-encrypt-set)) and use bucket policies to control access per user.

The key will add a second layer of protection and will also block operations that doesn't touch content. Listing, etc.

**Reply by avi-stx** [reply]
@klauspost 
can u pls explain more

**Reply by klauspost** [reply]
@avirosenberg See https://docs.min.io/minio/baremetal/security/encryption/encryption-key-management.html

Not sure what you want me to explain.
